File: web/src/models/team.ts
export type TeamView = {
  id: number;
  game_id: number;
  name: string;
  email: string | null;
  slogan: string | null;
  avatar_hash: string | null;
  has_writeup: boolean;
  state: State;
  pts: number;
  rank: number;
};

export enum State {
  Banned = 0,
  Preparing = 1,
  Pending = 2,
  Passed = 3,
}

File: crates/web/src/model/team.rs
use cds_db::{entity::team::State, sea_orm, sea_orm::FromQueryResult};
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, FromQueryResult)]
pub struct Team {
    pub id: i64,
    pub game_id: i64,
    pub name: String,
    pub email: Option<String>,
    pub slogan: Option<String>,
    pub state: State,
    pub pts: i64,
    pub rank: i64,
}
